api_server.py

```
@@ -549,4 +555,12 @@ def add_security_headers(response):
 """
 
 if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
+    # In production, this should be run with gunicorn.
+    # We only allow app.run() if FLASK_ENV is set to development.
+    if os.environ.get("FLASK_ENV") == "development":
+        app.run(host="0.0.0.0", port=int(os.environ.get("PORT", 5000)))
```

Ruff (0.14.14)

[error] Possible binding to all interfaces

(S104)